exporters/otlp/include/opentelemetry/exporters/otlp/otlp_environment.h

```diff
@@ -0,0 +1,203 @@
+  // Skip spaces
+  for (; *input && (' ' == *input || '\t' == *input || '\r' == *input || '\n' == *input); ++input)
+    ;
+
+  for (; *input && (*input >= '0' && *input <= '9'); ++input)
+  {
+    result = result * 10 + (*input - '0');
+  }
+
+  opentelemetry::nostd::string_view unit{input};
+  if ("ns" == unit)
+  {
+    return std::chrono::duration_cast<std::chrono::system_clock::duration>(
+        std::chrono::nanoseconds{result});
+  }
+  else if ("us" == unit)
+  {
+    return std::chrono::duration_cast<std::chrono::system_clock::duration>(
+        std::chrono::microseconds{result});
+  }
+  else if ("ms" == unit)
+  {
+    return std::chrono::duration_cast<std::chrono::system_clock::duration>(
+        std::chrono::milliseconds{result});
+  }
+  else if ("m" == unit)
+  {
+    return std::chrono::duration_cast<std::chrono::system_clock::duration>(
+        std::chrono::minutes{result});
+  }
+  else if ("h" == unit)
+  {
+    return std::chrono::duration_cast<std::chrono::system_clock::duration>(
+        std::chrono::hours{result});
+  }
+  else
+  {
+    return std::chrono::duration_cast<std::chrono::system_clock::duration>(
+        std::chrono::seconds{result});
+  }
+}
+
+inline const std::chrono::system_clock::duration GetOtlpDefaultTimeout()
+{
+  constexpr char kOtlpTracesTimeoutEnv[] = "OTEL_EXPORTER_OTLP_TRACES_TIMEOUT";
+  constexpr char kOtlpTimeoutEnv[]       = "OTEL_EXPORTER_OTLP_TIMEOUT";
+
+  auto timeout = opentelemetry::sdk::common::GetEnvironmentVariable(kOtlpTracesTimeoutEnv);
+  if (timeout.empty())
+  {
+    timeout = opentelemetry::sdk::common::GetEnvironmentVariable(kOtlpTimeoutEnv);
+  }
+  return GetOtlpTimeoutFromString(timeout.c_str());
+}
+
+struct cmp_ic
+{
+  bool operator()(const std::string &s1, const std::string &s2) const
+  {
+    return std::lexicographical_compare(
+        s1.begin(), s1.end(), s2.begin(), s2.end(),
+        [](char c1, char c2) { return ::tolower(c1) < ::tolower(c2); });
+  }
+};
+using OtlpHeaders = std::multimap<std::string, std::string, cmp_ic>;
+
+inline void DumpOtlpHeaders(OtlpHeaders &output, const char *env_var_name)
+{
+  auto value = opentelemetry::sdk::common::GetEnvironmentVariable(env_var_name);
+  if (value.empty())
+  {
+    return;
+  }
+
+  opentelemetry::common::KeyValueStringTokenizer tokenizer{value};
+  opentelemetry::nostd::string_view header_key;
+  opentelemetry::nostd::string_view header_value;
+  bool header_valid = true;
+
+  while (tokenizer.next(header_valid, header_key, header_value))
+  {
+    if (header_valid)
+    {
+      output.emplace(std::make_pair((std::string)header_key, (std::string)header_value));
+    }
+  }
+}
+
+inline OtlpHeaders GetOtlpDefaultHeaders()
+{
+  constexpr char kOtlpTracesHeadersEnv[] = "OTEL_EXPORTER_OTLP_TRACES_HEADERS";
+  constexpr char kOtlpHeadersEnv[]       = "OTEL_EXPORTER_OTLP_HEADERS";
+
+  OtlpHeaders result;
+  DumpOtlpHeaders(result, kOtlpHeadersEnv);
+  DumpOtlpHeaders(result, kOtlpTracesHeadersEnv);
```

Comment:
What should be the behavior in this scenario:

         OTEL_EXPORTER_OTLP_HEADERS = 'k1=v1,k2=v2'
         OTEL_EXPORTER_OTLP_TRACES_HEADERS = 'k1=v3'

As per the specs, the signal env variable overrides the other. which means, "k1=v3, k2=v2", and not "k1=v1, k2=v2,k1=v3".  Also if the metadata is explicitly specified, how should the merge looks like. Not blocking, but perhaps we can bring more clarify afterward.

Reply:
Thanks and I have already fixed it. I didn't find the document about the env variable override bdefore.